src/provider/controller/IntentHandler.ts

```diff
@@ -109,8 +109,12 @@ export class IntentHandler {
     }
 
     private async fireIntent(intent: Intent, appInfo: Application): Promise<IntentResolution> {
-        const appWindows = await this._model.findOrCreate(appInfo);
-        // to decide between focus nothing or apps with intent listener
+        await this._model.ensureRunning(appInfo);
+
+        // TODO: This will never resolve if no windows of this app connect to FDC3 [SERVICE-556]
```

Comment:
How concerned should we be about this possibility before 556 goes in? What affects will it have on the rest of the service?

Also the same question for the similar comments in other files.

Reply:
Ah, realised my " I don't think they're making anything worse" line was wrong. Have revised these so there's an explicit timeout around the `expectWindowsForApp` call

Reply:
Cool. I'd say that's good enough for now. As long as we aren't going to end up deadlocked.